pkg/tfbridge/walk/walk.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package walk
+
+import (
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/walk"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+)
+
+type SchemaPath = walk.SchemaPath
+
+// Translate a Pulumi property path to a bridged provider schema path.
+//
+// The function hides the complexity of mapping Pulumi property names to Terraform names, joining Schema with user
+// overrides in SchemaInfos, and accounting for MaxItems=1 situations where Pulumi flattens collections to plain values.
+// and therefore SchemaPath values are longer than the PropertyPath values.
+//
+// PropertyPathToSchemaPath may return nil if there is no matching schema found. This may happen when drilling down to
+// values of unknown type, attributes not tracked in schema, or when there is a type mismatch between the path and the
+// schema.
+func PropertyPathToSchemaPath(
+	propertyPath resource.PropertyPath,
+	schemaMap shim.SchemaMap,
+	schemaInfos map[string]*tfbridge.SchemaInfo) SchemaPath {
```

Comment:
Completely optional, but I think this is much easier to read.
```suggestion
	schemaInfos map[string]*tfbridge.SchemaInfo,
) SchemaPath {
```